grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -450,6 +455,10 @@ class StockMovementService {
             picklist.addToPicklistItems(picklistItem)
         }
 
+        if (reasonCode && requisitionItem.pickReasonCode != reasonCode) {
```

Comment:
Can you explain this logic? Please add a comment to the code.

Reply:
Comment added.